### PR TITLE
Add a gr0 exclusion for RA on isel extended mnemonics

### PR DESCRIPTION
**Pull request by aviansie-ben**
This PR adds code to prevent gr0 from being automatically assigned to the RA field of all extended mnemonics of `isel`, since these instructions will treat this as having a value of 0 regardless of the actual value stored in gr0. Additionally, the way in which gr0 exclusions are specified has been changed to use an instruction property flag, as proposed by #5297.

### CODE REVIEW

**Comment by gita-omr** [comment]
@zl-wang could you take a look as well?

**Comment by aviansie-ben** [comment]
@gita-omr To find instructions with this gr0 behaviour, I combined the existing lists that we had in the various `assignRegisters` functions with a list of all instructions which had the property bits marking them as a load or store set. This may not necessarily be exhaustive, but it matches our current behaviour.

**Comment by zl-wang** [comment]
1)  Wrong on:   addic
2) There is no distinction between taking value 0 vs. gr0 being invalid:
     For all update or indexed update form of load and store instructions, gr0 is invalid for RA, instead of taking it as value 0. You appeared to assume taking value 0 for all load/store.

**Comment by aviansie-ben** [comment]
>     1. Wrong on:   addic

Oops, you're right. This was meant to be applied to `addi2`, but somehow I ended up putting it on `addic` as well. Fixed in f028c5436dad753f2eb03d9202d5dd141fa02f76 (note to committers: I intend to squash this change after review is complete, before the PR is merged).

>     2. There is no distinction between taking value 0 vs. gr0 being invalid:
>        For all update or indexed update form of load and store instructions, gr0 is invalid for RA, instead of taking it as value 0. You appeared to assume taking value 0 for all load/store.

Hmm, you're right. In this case, a better name for this flag is needed, as it should only represent `gr0` avoidance in RA given how update instructions work. It took me a while to settle on this name, as it was quite hard for me to come up with a name that wasn't too wordy, but I'll see if I can come up with anything better. Do you have any suggestions?

**Comment by gita-omr** [comment]
> @gita-omr To find instructions with this gr0 behaviour, I combined the existing lists that we had in the various `assignRegisters` functions with a list of all instructions which had the property bits marking them as a load or store set. This may not necessarily be exhaustive, but it matches our current behaviour.

Could you please open a separate issue to make sure the set of opcodes with this flag is complete?

**Comment by zl-wang** [comment]
> Do you have any suggestions?

excludingGR0RA, or something like that

**Comment by aviansie-ben** [comment]
@zl-wang Changed the name of the flag to be more correct (and squashed some previous changes) in the latest force-push.

**Comment by aviansie-ben** [comment]
@gita-omr Opened #5361

**Comment by aviansie-ben** [comment]
Rebased to correct a merge conflict.

**Comment by fjeremic** [comment]
@genie-omr build aix,plinux
